assets/js/src/segments/dynamic/dynamic_segments_filters/custom_fields/text.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { useEffect } from 'react';
+import {
+  __,
+  assign,
+  compose,
+  get,
+  set,
+} from 'lodash/fp';
+
+import MailPoet from 'mailpoet';
+import Select from 'common/form/select/select';
+import Input from 'common/form/input/input';
+import { Grid } from 'common/grid';
+
+import {
+  WordpressRoleFormItem,
+  OnFilterChange,
+} from '../../types';
+
+interface Props {
+  onChange: OnFilterChange;
+  item: WordpressRoleFormItem;
+}
+
+export const Text: React.FunctionComponent<Props> = ({ onChange, item }) => {
+  useEffect(() => {
+    if (item.operator === undefined) {
+      onChange(
+        assign(item, { operator: 'equals', value: '' })
+      );
+    }
+  }, [onChange, item]);
+
+  return (
+    <>
+      <div className="mailpoet-gap" />
+      <Grid.CenteredRow>
+        <Select
+          key="select"
+          value={item.operator}
+          onChange={compose([
+            onChange,
+            assign(item),
+            set('operator', __, {}),
```

Comment:
Just to make sure I understand this. __ will be replaced by the value returned from previous function, correct? I tried to google it but the documentation for the library is either hard to find or non-existing.

Reply:
Correct. I know it from this [library](https://ramdajs.com/docs/#__). Here is something from [lodash](https://github.com/lodash/lodash/wiki/FP-Guide#placeholders)